### PR TITLE
Bacpop-160 network tab graphs

### PR DESCRIPTION
**Pull request by absternator**
This work entails showing all network graphs for a project when complete. THe following work is done 
- New endpoint in beebop & beebop_py to get all network graphs for project.... NOte: this beebop is targeting https://github.com/bacpop/beebop_py/pull/37 ... once emma has merged her stuff this will be editing with unit tests 
- FE work to display network tabs and be able to full screen to see each graph
image

image

### CODE REVIEW

**Review by EmmaLRussell — COMMENTED** [review]
Looks great! Couple of tiny comments

The network tab was not enabled until after the microreact task had finished - but it shouldn't need to wait for that.

I think the fullscreen functionality is probably too subtle for many uses to notice. The tooltip is good, but I think the icon should be on the graph panel itself - and could we use a standard fullscreen/expand icon (i.e. diagonal arrow heads to bottom left and top right, like in gmail). And/or could have additional instruction at top of tab.

We should order the clusters by numeric value really.

Fullscreen mode fits a bit awkwardly, and always seems to have an unexpected scrollbar.

It's really easy to lose the networks by scrolling in the panel - this was always an issue but it's now even harder to get them back now that the tab doesn't re-render on select. Can we make a ticket to either reset view or ensure the scrollaway can't happen in the first place?

File: app/client-v2/src/components/ProjectView/ProjectPostRun.vue [hunk 1]
```diff
@@ -46,7 +54,7 @@ const store = useProjectStore();
       </ProjectDataTable>
     </TabPanel>
     <TabPanel header="Network" :disabled="!store.isProjectComplete">
-      <NetworkTab />
+      <NetworkTab v-if="store.isProjectComplete && hasVisitedNetworkTab" />
```
[the remark this answers is not in the document]
Comment (absternator) [reply app/client-v2/src/components/ProjectView/ProjectPostRun.vue:57]: good catch thanks

File: app/client-v2/src/components/ProjectView/NetworkGraph.vue [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+const props = defineProps<{
+  graph: string;
+  cluster: string;
+}>();
+
+const fullScreenVisible = ref(false);
+</script>
+
+<template>
+  <Dialog v-model:visible="fullScreenVisible" modal>
```
[the remark this answers is not in the document]
Comment (absternator) [reply app/client-v2/src/components/ProjectView/NetworkGraph.vue:13]: yup that is right.. but have replaced to fix weird sizing issue

File: app/client-v2/patches/cytoscape-graphml+1.0.6.patch [hunk 3]
```diff
@@ -0,0 +1,34 @@
+diff --git a/node_modules/cytoscape-graphml/cytoscape-graphml.js b/node_modules/cytoscape-graphml/cytoscape-graphml.js
```
[the remark this answers is not in the document]
Comment (absternator) [reply app/client-v2/patches/cytoscape-graphml+1.0.6.patch:2]: added comment... permant until the package fixes the issue.. il create a package as well

File: app/client-v2/src/components/ProjectView/CytoscapeCanvas.vue [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    style: [
+      {
+        selector: "node",
+        style: {
+          width: "10px",
+          height: "10px",
+          content: "data(key0)",
+          "font-size": "7px",
+          color: "#00CC66",
+          "background-color": "rgba(45, 212, 191, 0.44)"
```
[the remark this answers is not in the document]
Comment (absternator) [reply app/client-v2/src/components/ProjectView/CytoscapeCanvas.vue:29]: sadly not i tried to initially

File: app/client-v2/src/components/ProjectView/CytoscapeCanvas.vue [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        }
+      },
+      {
+        selector: 'node[ref_query = "query"]',
+        style: {
+          "background-color": "crimson",
```
[the remark this answers is not in the document]
Comment (absternator) [reply app/client-v2/src/components/ProjectView/CytoscapeCanvas.vue:35]: once again sadly didnt work i tried lots before

**Comment by absternator** [comment]
> I think the fullscreen functionality is probably too subtle for many uses to notice. The tooltip is good, but I think the icon should be on the graph panel itself - and could we use a standard fullscreen/expand icon (i.e. diagonal arrow heads to bottom left and top right, like in gmail). And/or could have additional instruction at top of tab.
> 
> We should order the clusters by numeric value really.
> 
> Fullscreen mode fits a bit awkwardly, and always seems to have an unexpected scrollbar.
> 
> It's really easy to lose the networks by scrolling in the panel - this was always an issue but it's now even harder to get them back now that the tab doesn't re-render on select. Can we make a ticket to either reset view or ensure the scrollaway can't happen in the first place?

1. Have added outline so fullscreen easier to see and also made to be actually fullscreen (not dialog). The icon is the fullscreen icon from the library sadly
2. Wondering why numeric? they don't have any significance? also I'm just returning what we get from API.. 
3. il create ticket for graph reset

File: scripts/common [hunk 6]
```diff
@@ -1,2 +1,3 @@
-export API_BRANCH=main
+# TODO: change back to main before merge into main
+export API_BRANCH="bacpop-160-network-tab"
```
[the remark this answers is not in the document]
Comment (absternator) [reply scripts/common:2]: yup will do next time

File: app/client-v2/src/components/ProjectView/NetworkGraph.vue [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+const props = defineProps<{
+  graph: string;
+  cluster: string;
+}>();
+
+const fullScreenVisible = ref(false);
+</script>
+
+<template>
+  <Sidebar v-model:visible="fullScreenVisible" :header="`Cluster: ${props.cluster}`" position="full">
```
Comment (EmmaLRussell) [inline app/client-v2/src/components/ProjectView/NetworkGraph.vue:13]: This does feel like a weird use of a sidebar, but I guess it gets the job done. 
I wonder if there's accessibility issues... but then the network isn't exactly accessible anyway!

**Review by EmmaLRussell — APPROVED** [review]
Looks great, thanks!

I do still get a scrollbar in fullscreen mode though.. 
Also noticed there's zero padding between the bottom of the fullscreen button and the top of the graph panel in fullscreen mode. 

Tiny suggestion below. 

But these are all super minor so happy to approve.

**Pull request merged**